crates/top/re_sdk/src/lenses/ast.rs

```diff
@@ -174,266 +217,270 @@ impl Lens {
             return Default::default();
         };
 
-        let mut builders = ahash::HashMap::default();
-        for output in &self.component_columns {
-            let entity_path = output
-                .new_entity_path
-                .as_ref()
-                .unwrap_or(chunk.entity_path());
-
-            let components = builders
-                .entry((entity_path.clone(), output.is_static))
-                .or_insert_with(ChunkComponents::default);
-
-            if components.contains_key(&output.component_descr.component) {
-                re_log::warn_once!("Replacing duplicated component {}", output.component_descr);
-            }
-
-            let mut list_array_result = column.list_array.clone();
-            for op in &output.ops {
-                match op.call(&list_array_result) {
-                    Ok(result) => {
-                        list_array_result = result;
-                    }
-                    Err(err) => {
-                        re_log::error!(
-                            "Lens operation '{op:?}' failed for output column '{}' on entity '{entity_path}': {err}",
-                            output.component_descr.component,
-                        );
-                        return vec![];
-                    }
-                }
-            }
-
-            components.insert(SerializedComponentColumn::new(
-                list_array_result,
-                output.component_descr.clone(),
-            ));
-        }
-
-        let mut time_builders: ahash::HashMap<
-            &EntityPath,
-            HashMap<TimelineName, re_chunk::TimeColumn>,
-        > = ahash::HashMap::default();
-        for time in &self.time_columns {
-            let entity_path = time.new_entity_path.as_ref().unwrap_or(chunk.entity_path());
-
-            let time_entry = time_builders.entry(entity_path).or_default();
+        self.outputs
+            .iter()
+            .filter_map(|output| match output {
+                LensKind::OneToOne {
+                    target_entity,
+                    components,
+                    timelines,
+                } => apply_one_to_one(chunk, column, target_entity, components, timelines),
```

Comment:
Here's an example of where using structs for the variants would lead to cleaner code
```suggestion
                LensKind::OneToOne(one_to_one) => one_to_one.apply(chunk, column),
```